src/main/java/com/foodybuddy/service/impl/CityServiceImpl.java

```diff
@@ -0,0 +1,289 @@
+
+	/** The session factory. */
+	private SessionFactory sessionFactory;
+
+	/**
+	 * Instantiates a new city service impl.
+	 *
+	 * @param sessionFactory
+	 *            the session factory
+	 */
+	public CityServiceImpl(SessionFactory sessionFactory) {
+		if (sessionFactory == null) {
+			throw ObjectNullException;
+		}
+		this.sessionFactory = sessionFactory;
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see com.foodybuddy.service.CityService#getById(java.lang.Integer)
+	 */
+	public City getById(Integer id) throws Exception {
+		
+		Session session = null;
+		try {
+			if (id == null || id <= 0) {
+				if (id == null) {
+					throw new RuntimeException("ID is null");
+				} else {
+					throw new RuntimeException("ID is invalid");
+				}
+			}
+			
+			session = this.sessionFactory.openSession();
+			CityDAO cityDAO = new CityDAOImpl(session);
+			return cityDAO.getById(id);
+			
+		} catch (Exception exception) {
+			throw exception;
+		} finally {
+			if (session != null) {
+				session.close();
+			}
+		}
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see com.foodybuddy.service.CityService#getAll()
+	 */
+	public List<City> getAll() throws Exception {
+		Session session = null;
+		try {
+			session = this.sessionFactory.openSession();
+			CityDAO cityDAO = new CityDAOImpl(session);
+			List<City> cityList = cityDAO.getAll();
+			if (cityList.size() == 0)
+				throw new RuntimeException("Empty Country List");
+			return cityList;
+		} catch (Exception exception) {
+			throw exception;
+		} finally {
+			if (session != null) {
+				session.close();
+			}
+		}
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see com.foodybuddy.service.CityService#insert(java.lang.String,
+	 * java.lang.Integer)
+	 */
+	public City insert(String name, Integer stateId) throws TransactionException {
+
+		Session session = null;
+		Transaction transaction = null;
+
+		try {
+			
+			if (name == null || stateId == null || stateId <= 0 || name.trim().length() == 0) {
+
+				if (name == null) {
+					throw new RuntimeException("Name is null");
+				}
+
+				else if (stateId == null) {
+					throw new RuntimeException("StateId is null");
+				}
+
+				else if (stateId <= 0) {
+					throw new RuntimeException("StateId is null");
+				}
+
+				else {
+					throw new RuntimeException("Name is invalid");
+				}
+			}
+			session = this.sessionFactory.openSession();
+			transaction = session.beginTransaction();
+			transaction.setTimeout(10);
+
+			StateDAO stateDAO = new StateDAOImpl(session);
+			State state = stateDAO.getById(stateId);
+
+			if (state == null) {
+				throw ObjectNullException;
+			}
+
+			City city = new City();
+			city.setName(name);
+			city.setState(state);
+
+			CityDAO cityDAO = new CityDAOImpl(session);
+			cityDAO.insert(city);
+			transaction.commit();
+			return city;
+		}
+
+		catch (Exception exception) {
+			try {
+				if (transaction != null)
+					transaction.rollback();
+				throw TransactionFailureException;
```

Comment:
Same type of exception is thrown even if rollback is successfull or not which should not be the case